workshop/README.md

```diff
@@ -125,35 +161,45 @@ and stores this data in an in-memory key-value store inside the Lisk application
+Notice that we return an empty array at the end of the function. The same idea applies here as well. As the `validateAsset()` function does only allow for static checks, more advanced checks that require data from the store can be performed in the `applyAsset()` function. In case of an error, we put the error in an array and return this array. However, for the `invoiceTransaction`, we do not require additional validation steps, so we decided to simply return an empty array.<br><br>
 
 #### undoAsset()
-Now it's your turn. Your task is to write the reversed logic of the `applyAsset()` function. The `undoAsset()` function is used to roll back changes that were done by the `applyAsset()` function.
+Now, it's your turn. Your task is to write the reversed logic of the `applyAsset()` function. The `undoAsset()` function is used to roll back changes that were done by the `applyAsset()` function.
 
-At this moment, the function is called when a fork occurs and we want to switch to a different chain. In that case, we have to undo transactions and we need to know how to undo a transaction for each specific transaction type. That's why we have to define custom undo logic for this custom transaction.
+**Use case:** At this moment, the function is called when a fork occurs and we want to switch to a different chain. In that case, we have to undo transactions and we need to know how to undo a transaction for each specific transaction type. That's why we have to define custom undo logic for this custom transaction.
 
-**Task 4: Complete code for undo function:**
+**Task 4: Complete the code for Undo() function:**
+Here's the flow of actions for the `undo()` function:
 1. Retrieve sender account
-2. Reduce `invoiceCount` (remember the `undefined` state when it's the first transaction)
-3. Remove id from `invoicesSent` array (Tip: use splice - Also remember `undefined` case)
-4. Return errors
+2. Reduce `invoiceCount`
+3. Remove id from `invoicesSent` array
+4. Save updated sender object
 
-_The solution can be found in the `Solution: Invoice Transaction` section._
+_The solution can be found in the `Solution: Invoice Transaction` section._<br><br>
 
 ### Testing the InvoiceTransaction
 To test the `InvoiceTransaction`, we have to register the custom transaction to our blockchain in the `index.js` file in the root of the `/workshop` folder.
 
 You'll find code that is commented out. Uncomment the line for importing the `InvoiceTransaction` and uncomment the line of code that registers the custom transaction to the application.
 
-Next, verify if everything is fine by starting the application with `npm start`. In order to verify if our custom invoice transaction works, we should send an invoice transaction. The `/workshop/generator/invoice.js` file contains a generator that uses a genesis account with sufficient funds and creates a JSON object (Run `node generator/invoice.js`).
+![Register custom transactoin](./assets/01-index-register.png)
```

Comment:
Why are we using an image instead of a code snippet here?

Reply:
Wanted to add some visuals to make the text bit more readable/segregated.